applications/nrf5340_audio/src/bluetooth/bt_rendering_and_capture/volume/bt_vol_ctlr.c

```diff
@@ -70,49 +98,149 @@ static bool vcs_client_peer_exists(struct bt_conn *conn)
 }
 
 /**
- * @brief	Callback handler for the volume state.
+ * @brief Get the index of a given Volume Control Server.
+ *
+ * @param[in] vcs The given volume control server.
+ * @return int 0 or positive integer on success, -EFAULT if not found.
+ */
+static int vcs_peer_index_get(struct bt_vcp_vol_ctlr const *const vcs)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+		if (!vcs_peer_alive(vcs)) {
+			/* Skip */
+			continue;
+		}
+
+		if (vcs == vcs_client_peer[i].peer) {
+			return i;
+		}
```

Comment:
If you clear the `vcs_client_peer[i].peer` on disconnects, you could skip the `vcs_peer_alive` check and simply do `vcs_peer_alive[i].peer == NULL`